src/ProducerEventProcessor/ProducerEventProcessor.cpp

```diff
@@ -270,6 +322,11 @@ void ProducerEventProcessorImpl::ProcessCallstackSampleAndTransferOwnership(
 
 void ProducerEventProcessorImpl::ProcessCaptureFinishedAndTransferOwnership(
     CaptureFinished* capture_finished) {
+  if (!thread_state_slice_tid_and_begin_timestamp_to_callstack_.empty()) {
+    ORBIT_ERROR(
```

Comment:
How likely is this too happen?

Reply:
This is very unlikely. We were about to add a check, but decided to not crash the service in that case, as it does not hurt us.

Reply:
Totally agree on not having a check, but what do you mean with "unlikely"? Do we know of a situation where this can happen?

Reply:
I can not think of any. It would basically mean, that the SwitchesNamesStateVisitor did not produce a slice for the begin tracepoint, even including the "clean up at the end".

Reply:
Sounds good. Maybe you can add a quick comment like
"We don't expect this to happen because SwitchesNamesStateVisitor always produces a slice from the remaining begin tracepoints at the end of the capture."

Reply:
Done.